### PR TITLE
IA-2566: Embedded links - user field : Value not found when the current user has not user permission

### PR DESCRIPTION
**Pull request by hakifran**
Explain what problem this PR is resolving
- When the current user has no **User management - Admin** the userSelect in embedded link display **a no found error**
- This ticket is about to fix that
Related JIRA tickets : [IA-2566](https://bluesquare.atlassian.net/browse/IA-2566)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
- Give the possibility to define the value of managedUsersOnly parameter instead of the predefined "true" value
## How to test
- Remove the **User management - Admin** permission from one user
- Connect with the user and try to create a new embedded link or edit one and check if there is no an error in userSelect field
## Print screen / video
[Screencast from 2024-01-08 10-06-40.webm](https://github.com/BLSQ/iaso/assets/19631540/ed5ae061-ead3-4056-b22e-d733b77a4d3c)

[IA-2566]: https://bluesquare.atlassian.net/browse/IA-2566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hakifran** [comment]
@cypress

File: hat/assets/js/apps/Iaso/domains/users/hooks/useGetProfiles.js [hunk 1]
```diff
@@ -19,19 +17,21 @@ export const useGetProfilesApiParams = params => {
...
+              managedUsersOnly: params.managedUsersOnly,
           }
         : {};
 
     const url = makeUrlWithParams(`/api/profiles/`, {
         ...apiParams,
-        managedUsersOnly: 'true',
+        managedUsersOnly: apiParams.managedUsersOnly
+            ? apiParams.managedUsersOnly
+            : 'true',
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/hooks/useGetProfiles.js]:
```suggestion
        managedUsersOnly: apiParams.managedUsersOnly ?? 'true',
```

nitpick

**Review by quang-le — CHANGES_REQUESTED** [review]
I still get an error in a polio page. I don't know if there's anything special with user id 1
Screenshot 2024-01-08 at 12 37 16

Otherwise not much to say on the front-end side. I feel you could spare addinga prop and just pass false to `useGetProfiles` since both the dialog and the user select are custom for the embedded link (so they're not used anywhere else).

I have no clue what the backend does, but I guess there was a separate PR for that.

**Comment by hakifran** [comment]
@cypress

**Comment by hakifran** [comment]
> I still get an error in a polio page. I don't know if there's anything special with user id 1 Screenshot 2024-01-08 at 12 37 16
> 
> Otherwise not much to say on the front-end side. I feel you could spare addinga prop and just pass false to `useGetProfiles` since both the dialog and the user select are custom for the embedded link (so they're not used anywhere else).
> 
> I have no clue what the backend does, but I guess there was a separate PR for that.

@quang-le the user with 1 as id has not a profile linked to it that's why it generates the error. I will talk with Martin to see how to handle situations like this.

**Comment by hakifran** [comment]
> I still get an error in a polio page. I don't know if there's anything special with user id 1 Screenshot 2024-01-08 at 12 37 16
> 
> Otherwise not much to say on the front-end side. I feel you could spare addinga prop and just pass false to `useGetProfiles` since both the dialog and the user select are custom for the embedded link (so they're not used anywhere else).
> 
> I have no clue what the backend does, but I guess there was a separate PR for that.

In the backend there is a filter on managedUserOnly for profiles, so what I did is to skip this filter so that I can get the list of all users
Screenshot from 2024-01-09 11-34-18

File: hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx [hunk 2]
```diff
@@ -62,7 +64,9 @@ export const UsersSelect: FunctionComponent<Props> = ({
             loading={isFetchingProfiles}
             clearable={false}
             multi
-            value={value}
+            value={value?.filter(userId =>
+                profilesList.map(i => i.value).includes(userId),
+            )}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/components/UsersSelect.tsx]:
```suggestion
            value={value?.filter(userId =>
                Boolean(profilesList.find(profile => profile.value===userId))
            )}
```

**Review by quang-le — APPROVED** [review]
juste a small optimization suggestion

**Pull request merged**